### PR TITLE
Documentation update for batching, pipeline capacity, streams eviction

### LINKED ISSUE #210: Document pipeline params 'batched_push_timeout', 'max_parallel_streams'
Currently undocumented.

### PR DESCRIPTION
**Pull request by bwsw**
Closes #210
Documentation for #224

### CODE REVIEW

File: docs/source/advanced_topics/0_pipeline_capacity.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Pipeline Capacity
+====================
+
+Savant allows setting the pipeline capacity parameter to define how many streams a pipeline can simultaneously process. The parameter is defined with the ``max_parallel_streams`` property in the general parameters section. When the number of streams directed to the pipeline is higher than the value set by the parameter, the pipeline will terminate with the error.
+
+This parameter is set to ``64`` by default, a pretty high value for the real-time pipeline. Consider this parameter as a trigger determining the cases when you wrongly directed more streams to the pipeline than it is expected to handle on particular hardware. Keep the pipeline reasonably low for real-time pipelines to not break real-time constraints. In contrast, you may want to increase the parameter for non-real-time pipelines to ensure that the pipeline meets sharding expectations.
```
Comment (abramov-oleg) [inline docs/source/advanced_topics/0_pipeline_capacity.rst]: `Keep the pipeline reasonably low for real-time pipelines to not break real-time constraints.` ->
`Keep the parameter reasonably low for real-time pipelines to not break real-time constraints.`

File: docs/source/advanced_topics/0_pipeline_capacity.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+It is important to remember that in certain situations, the pipeline can technically experience more simultaneous streams than expected, e.i., when one stream is dead but not yet evicted, and another appears. Knowing that, set the parameter slightly higher to avoid pipeline termination in transition situations.
```
Comment (abramov-oleg) [inline docs/source/advanced_topics/0_pipeline_capacity.rst]: As I understand, `e.i.` means `for example` in the context. If so the correct variant is `e.g.`

File: docs/source/advanced_topics/0_dead_stream_eviction.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
+Dead Streams Eviction
+=====================
+
+Savant automatically evicts dead streams. You know about such evictions by receiving corresponding events (via designated callbacks) in your Pyfuncs: normally, you do not need to deal with evictions somehow. Instead, configuration parameters specified per source tune how frequently Savant checks the idle streams and how long the stream can be inactive before being considered dead.
```
Comment (abramov-oleg) [inline docs/source/advanced_topics/0_dead_stream_eviction.rst]: ` normally, you do not need to deal with evictions somehow` -> `normally, you do not need to manually deal with evictions`

File: docs/source/advanced_topics/0_batching.rst [hunk 4]
```diff
@@ -1,4 +1,40 @@
 Batching
 ========
 
-TODO
\ No newline at end of file
+Batching is an important function influencing the performance of the pipeline a lot. Inference units utilize batching to process images more efficiently. Primary models (which operate on the whole image) consider the batch as a number of frames simultaneously passed to the model. Secondary models consider the batch as a number of frames simultaneously passed to the model.
```
Comment (abramov-oleg) [inline docs/source/advanced_topics/0_batching.rst]: `Secondary models consider the batch as a number of frames simultaneously passed to the model.` -> `Secondary models consider the batch as a number of objects simultaneously passed to the model.`

File: docs/source/advanced_topics/0_batching.rst [hunk 5]
```diff
@@ -1,4 +1,40 @@
...
+
+The pipeline configuration file has parameters that regulate the batch size. They are ``batch_size`` and ``batched_push_timeout``.
+
+The ``batch_size`` parameter may be configured for the first inference element or globally. However, the same value must be configured for the first inference element when it is configured globally. The global configuration is usually necessary when the pipeline doesn't include inference elements; however, it benefits from batching somehow. E.g., if the code uses asynchronous cuda operations, batched processing may result in better performance. However, usually, you must specify the ``batch_size`` parameter for the first inference element. By default, the value for the parameter is set to ``1``.
```
Comment (abramov-oleg) [inline docs/source/advanced_topics/0_batching.rst]: `somehow` -> `anyway`

**Review by abramov-oleg — CHANGES_REQUESTED** [review]

**Comment by bwsw** [comment]
@abramov-oleg fixed.

**Review by abramov-oleg — APPROVED** [review]

**Pull request merged**